### PR TITLE
Update sRGB API for Texture and RenderTarget

### LINKED ISSUE #222: Add proper support for sRGB
Due to the way the CSFML API was implemented for creating/loading textures, there was no way to create textures in sRGB mode and the exposed setSrgb method was essentially useless.

With https://github.com/SFML/CSFML/pull/177 this was fixed, which then however leaves a gap in SFML.NET. The property for sRGB needs to be adapted and the new creation functions need to be called from SFML.NET.

### PR DESCRIPTION
**Pull request by Marioalexsan**
Fixes #222, supersedes #236.

As far as I'm aware, most of the functionality for sRGB was already added through other PRs.

### CODE REVIEW

File: src/SFML.Graphics/RenderTarget.cs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+using System.Reflection;
```
Comment (eXpl0it3r) [inline src/SFML.Graphics/RenderTarget.cs:1]: Don't think we need Reflection here.

File: src/SFML.Graphics/RenderTarget.cs [hunk 2]
```diff
@@ -16,6 +17,14 @@ public interface RenderTarget
         ////////////////////////////////////////////////////////////
         Vector2u Size { get; }
 
+
+        ////////////////////////////////////////////////////////////
+        /// <summary>
+        /// Tell if the render target will use sRGB encoding when drawing on it
+        /// </summary>
+        ////////////////////////////////////////////////////////////
+        bool IsSrgb { get; }
```
Comment (eXpl0it3r) [inline src/SFML.Graphics/RenderTarget.cs:26]: Where is this set?

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]

File: src/SFML.Graphics/RenderTarget.cs (hunk 1 above)
Comment (Marioalexsan) [reply src/SFML.Graphics/RenderTarget.cs]: Removed it.

File: src/SFML.Graphics/RenderTarget.cs (hunk 2 above)
Comment (eXpl0it3r) [reply src/SFML.Graphics/RenderTarget.cs:25]: Ah my bad, I didn't realized this is on RenderTarget and that RenderTarget is an interface 😄

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
Thank you for aligning this! 🙂

**Pull request merged**